content/doc/book/installing/_docker.adoc

```diff
@@ -13,41 +13,135 @@ _run-jenkins-in-docker.adoc).
+  daemon and persist the Jenkins data using
+  the following
+  link:https://docs.docker.com/engine/reference/commandline/volume_create/[`docker volume create`]
+  commands:
++
+[source,bash]
+----
+docker volume create jenkins-docker-certs
+docker volume create jenkins-data
+----
+. In order to execute Docker commands inside Jenkins nodes, download and run
+  the `docker:dind` Docker image using the following
+  link:https://docs.docker.com/engine/reference/commandline/container_run/[`docker container run`]
+  command:
++
+[source,bash]
+----
+docker container run \
+  --name jenkins-docker \ # <1>
+  --rm \ # <2>
+  --detach \ # <3>
+  --privileged \ # <4>
+  --network jenkins \ # <5>
+  --network-alias docker \ # <6>
+  --env DOCKER_TLS_CERTDIR=/certs \ # <7>
+  --volume jenkins-docker-certs:/certs/client \ # <8>
+  --volume jenkins-data:/var/jenkins_hom \ # <9>
+  --publish 2376:2376 \ # <10>
+  docker:dind # <11>
+----
+<1> ( _Optional_ ) Specifies the Docker container name to use for running the
+image. By default, Docker will generate a unique name for the container.
+<2> ( _Optional_ ) Automatically removes the Docker container (the instance of
+the Docker image) when it is shut down. This contains the Docker image cache
+used by Docker when invoked from the `jenkinsci/blueocean` container described
+below.
+<3> ( _Optional_ ) Runs the Docker container in the background. This instance
+can be stopped later by running `docker container stop jenkins-docker` and
+started again with `docker container start jenkins-docker`. See
+link:https://docs.docker.com/engine/reference/commandline/container/[`docker container`]
+for more container management commands.
+<4> Running Docker in Docker currently requires privileged access to function
+properly. This requirement may be relaxed with newer Linux kernel versions.
+// TODO: what versions of Linux?
+<5> This corresponds with the network created in the earlier step.
+<6> Makes the Docker in Docker container available as the hostname `docker`
+within the `jenkins` network.
+<7> Enables the use of TLS in the Docker server. Due to the use
+of a privileged container, this is recommended, though it requires the use of
+the shared volume described below. This environment variable controls the root
+directory where Docker TLS certificates are managed.
+<8> Maps the `/certs/client` directory inside the container to
+a Docker volume named `jenkins-docker-certs` as created above.
+<9> Maps the `/var/jenkins_home` directory inside the container to the Docker
+volume named `jenkins-data` as created above. This will allow for other Docker
+containers controlled by this Docker container's Docker daemon to mount data
+from Jenkins.
+<10> Exposes the Docker daemon port so that other Docker containers can connect
+to it to execute `docker` commands. This port is the default Docker TLS port.
+Note that other containers will need to configure the `DOCKER_HOST` environment
+variable to use this daemon.
```

Comment:
Due to the problem I got with the port, I tried without publishing the 2376 port and it works fine. It seems that the `tcp://docker` url, is resolved internally to the docker engine, not something on the host, and thus, the port is automatically open with 2376 inside the defined docker network.

Reply:
Thanks for figuring this out! I've removed the port publishing from the Windows commands, and I've marked them optional for the macOS/Linux ones.